logs/command.go

```diff
@@ -2,104 +2,59 @@ package logs
-		fmt.Printf("    %2s, %16s %70s\n", "-h", "--help", "Show usage")
-		fmt.Printf("    %2s  %16s %70s\n", "", "--min-time MIN", "Earliest time to search from")
-		fmt.Printf("    %2s  %16s %70s\n", "", "--max-time MAX", "Latest time to search from")
-		fmt.Printf("    %2s, %16s %70s\n", "-c", "--configfile", "Path to config (~/.swo-cli.yaml)")
-		fmt.Printf("    %2s, %16s %70s\n", "-g", "--group GROUP_ID", "Group ID to search")
-		fmt.Printf("    %2s, %16s %70s\n", "-s", "--system SYSTEM", "System to search")
-		fmt.Printf("    %2s, %16s %70s\n", "-j", "--json", "Output raw JSON data (off)")
-		fmt.Printf("    %2s, %16s %70s\n", "-V", "--version", "Display the version and exit")
-
-		fmt.Println()
-
-		fmt.Println("    Usage:")
-		fmt.Println("      swo-cli logs [--min-time time] [--max-time time] [-g group-id] [-s system]")
-		fmt.Println("        [-c swo-cli.yml] [-j] [--] [query]")
-
-		fmt.Println()
-
-		fmt.Println("    Examples:")
-		fmt.Printf("    %s logs something\n", os.Args[0])
-		fmt.Printf("    %s logs 1.2.3 Failure\n", os.Args[0])
-		fmt.Printf(`    %s logs -s ns1 "connection refused"%v`, os.Args[0], "\n")
-		fmt.Printf(`    %s logs "(www OR db) (nginx OR pgsql) -accepted"%v`, os.Args[0], "\n")
-		fmt.Printf(`    %s logs -g <SWO_GROUP_ID> "(nginx OR pgsql) -accepted"%v`, os.Args[0], "\n")
-		fmt.Printf(`    %s logs --min-time 'yesterday at noon' --max-time 'today at 4am' -g <SWO_GROUP_ID>%v`, os.Args[0], "\n")
-		fmt.Printf("    %s logs -- -redis\n", os.Args[0])
+func run(cCtx *cli.Context) error {
+	opts := &Options{
+		args:       cCtx.Args().Slice(),
+		configFile: cCtx.String("config"),
+		group:      cCtx.String("group"),
+		system:     cCtx.String("system"),
+		maxTime:    cCtx.String("max-time"),
+		minTime:    cCtx.String("min-time"),
+		json:       cCtx.Bool("json"),
+		ApiUrl:     cCtx.String("api-url"),
+		Token:      cCtx.String("api-token"),
 	}
-
-	cmd.fs.StringVar(&cmd.opts.configFile, "c", "", "")
-	cmd.fs.StringVar(&cmd.opts.configFile, "configfile", defaultConfigFile, "")
-	cmd.fs.StringVar(&cmd.opts.group, "g", "", "")
-	cmd.fs.StringVar(&cmd.opts.group, "group", "", "")
-	cmd.fs.StringVar(&cmd.opts.system, "s", "", "")
-	cmd.fs.StringVar(&cmd.opts.system, "system", "", "")
-	cmd.fs.StringVar(&cmd.opts.ApiUrl, "api-url", defaultApiUrl, "")
-	cmd.fs.StringVar(&cmd.opts.minTime, "min-time", "", "")
-	cmd.fs.StringVar(&cmd.opts.maxTime, "max-time", "", "")
-	cmd.fs.BoolVar(&cmd.opts.json, "j", false, "")
-	cmd.fs.BoolVar(&cmd.opts.json, "json", false, "")
-	cmd.fs.BoolVar(&cmd.opts.version, "V", false, "")
-	cmd.fs.BoolVar(&cmd.opts.version, "version", false, "")
-
-	return cmd
-}
-
-func (c *command) Init(args []string) error {
-	err := c.fs.Parse(args)
-	if err != nil {
-		return err
-	}
-
-	opts, err := c.opts.Init(c.fs.Args())
+	opts.Init(cCtx.Args().Slice())
+	client, err := NewClient(opts)
 	if err != nil {
 		return err
 	}
 
-	client, err := NewClient(opts)
+	err = client.Run(context.Background())
```

Comment:
I think we should pass a context that is possible to be canceled.

Something like the one that was in old `main.go`:
```
ctx, cancel := context.WithCancel(context.Background())
c := make(chan os.Signal, 1)
signal.Notify(c, os.Interrupt)
defer func() {
	signal.Stop(c)
	cancel()
}()
```

Reply:
It's actually not necessary, unless there's some specific cleanup actions that you want. You can ^C the process and it terminates without this. Was there specifically some action that you'd prefer to be taken or am I missing something?

```
➜  swo-cli git:(grepory/cli) ./swo logs --min-time '1 hour ago' --max-time 'now' 'swo-k8s-collector'
^C
➜  swo-cli git:(grepory/cli)
```

Reply:
oh ok, my bad, it seems like it's not necessary.